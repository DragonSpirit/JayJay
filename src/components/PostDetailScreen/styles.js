import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  scrollViewStyle: {
    paddingHorizontal: '5%',
  },
  scrollViewContentStyle: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    flex: 0.9,
    fontSize: 17,
    textAlign: 'center',
    paddingVertical: 10,
  },
  commentButton: {
    width: '100%',
    marginVertical: 10,
  },
  headerContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  toggleFavoriteIcon: {
    flex: 0.1,
  },
})

export default styles
