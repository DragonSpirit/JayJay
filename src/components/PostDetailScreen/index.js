// @flow

import * as React from 'react'
import { View, ScrollView, Text, Share } from 'react-native'
import { Button, Icon } from 'react-native-elements'
import { ANIMATIONS_SLIDE, CustomTabs } from 'react-native-custom-tabs'
import HTMLView from 'react-native-htmlview'
import styles from './styles'
import { commonStyles } from '../common.styles'

type Props = {
  navigation: *,
  isFavorite: (id: number) => boolean,
  addToFavorite: (id: number) => void,
  removeFromFavorite: (id: number) => void,
}

class PostDetailScreen extends React.PureComponent<Props> {
  static navigationOptions = {
    title: 'Details',
  }

  static defaultProps = {
    addToFavorite: () => {},
    removeFromFavorite: () => {},
    isFavorite: () => false,
  }

  constructor(props: Props) {
    super(props)
  }

  openComments = () => {
    const { navigation } = this.props
    const item = navigation && navigation.state && navigation.state.params
    CustomTabs.openURL(`${item.url}#comments`, {
      toolbarColor: '#607D8B',
      enableUrlBarHiding: true,
      showPageTitle: true,
      enableDefaultShare: true,
      animations: ANIMATIONS_SLIDE,
      headers: {
        'my-custom-header': item.title,
      },
      forceCloseOnRedirection: true,
    })
  }

  openLink = (link: string) => {
    CustomTabs.openURL(link, {
      toolbarColor: '#607D8B',
      enableUrlBarHiding: true,
      showPageTitle: true,
      enableDefaultShare: true,
      animations: ANIMATIONS_SLIDE,
      forceCloseOnRedirection: true,
    })
  }

  sharePost = async () => {
    const { navigation } = this.props
    const item = navigation && navigation.state && navigation.state.params
    await Share.share({
      message: item.url,
      title: item.title,
    })
  }

  toggleFavorite = (id: number) => {
    const { isFavorite, addToFavorite, removeFromFavorite } = this.props
    if (isFavorite(id)) {
      removeFromFavorite(id)
    } else {
      addToFavorite(id)
    }
  }

  render() {
    const { navigation, isFavorite } = this.props
    const item = navigation && navigation.state && navigation.state.params
    return item ? (
      <View style={commonStyles.flex}>
        <View style={styles.headerContainer}>
          <Text style={styles.title}>{item.title}</Text>
          <View style={styles.iconsContainer}>
            <Icon
              name='heart'
              type='font-awesome'
              underlayColor={'transparent'}
              style={styles.toggleFavoriteIcon}
              color={isFavorite(item.id) ? '#184fff' : '#000'}
              onPress={() => {
                this.toggleFavorite(item.id)
              }}
            />
            <Icon
              name='share'
              type='font-awesome'
              underlayColor={'transparent'}
              style={styles.shareIcon}
              color='#000'
              onPress={this.sharePost}
            />
          </View>
        </View>
        <ScrollView style={[commonStyles.flex, styles.scrollViewStyle]}>
          <HTMLView
            value={item.text}
            onLinkPress={link => {
              this.openLink(link)
            }}
            addLineBreaks={false}
          />
        </ScrollView>
        <Button
          title='Комментарии'
          onPress={this.openComments}
          buttonStyle={[commonStyles.buttonStyleCommon, styles.commentButton]}
        />
      </View>
    ) : null
  }
}

export default PostDetailScreen
