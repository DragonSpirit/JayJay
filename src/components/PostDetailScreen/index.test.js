import React from 'react';
import PostDetailScreen from './';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<PostDetailScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
