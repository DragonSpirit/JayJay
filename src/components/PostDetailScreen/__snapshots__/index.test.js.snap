// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetailScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 0.9,
          "fontSize": 17,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Test
    </Text>
    <View
      style={
        Object {
          "flex": 0.15,
          "flexDirection": "column",
          "padding": 5,
        }
      }
    >
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingHorizontal": "5%",
        },
      ]
    }
  >
    <View>
      <View>
        <Text
          style={
            Array [
              null,
              null,
            ]
          }
        >
          Test
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(92, 99,216, 1)",
            "borderColor": "transparent",
            "borderRadius": 5,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 10,
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Комментарии
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`PostDetailScreen renders favorite post correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 0.9,
          "fontSize": 17,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Test
    </Text>
    <View
      style={
        Object {
          "flex": 0.15,
          "flexDirection": "column",
          "padding": 5,
        }
      }
    >
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#184fff",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingHorizontal": "5%",
        },
      ]
    }
  >
    <View>
      <View>
        <Text
          style={
            Array [
              null,
              null,
            ]
          }
        >
          Test
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(92, 99,216, 1)",
            "borderColor": "transparent",
            "borderRadius": 5,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 10,
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Комментарии
        </Text>
      </View>
    </View>
  </View>
</View>
`;
